From a6c3c48148dd60a09ffab57019f163180c159d65 Mon Sep 17 00:00:00 2001
Message-ID: <a6c3c48148dd60a09ffab57019f163180c159d65.1769383625.git.sam@gentoo.org>
From: Sandra Loosemore <sloosemore@baylibre.com>
Date: Sun, 25 Jan 2026 22:43:58 +0000
Subject: [PATCH] Fix gcc-urlifier selftest failure

My recent commits for PR122243 added index entries for -fno-* options
as well as their normal positive forms.  Apparently the "urlifier"
used to insert option URLS into diagnostic messages can find the
anchor for either form, but its self-tests are hard-wired to match
only the positive form for the two specific options it's looking up.
This patch robustifies it to allow it to match the anchor for either
the positive or negative forms.

gcc/ChangeLog
	* gcc-urlifier.cc (test_gcc_urlifier): Match either positive
	or negative option URLS.
---
 gcc/gcc-urlifier.cc | 14 ++++++++++----
 1 file changed, 10 insertions(+), 4 deletions(-)

diff --git a/gcc/gcc-urlifier.cc b/gcc/gcc-urlifier.cc
index d6958e3f6f86..37f87d55a915 100644
--- a/gcc/gcc-urlifier.cc
+++ b/gcc/gcc-urlifier.cc
@@ -262,12 +262,18 @@ test_gcc_urlifier ()
        doc_urls[idx].url_suffix);
 
   /* Check an option.  */
-  ASSERT_STREQ (u.get_url_suffix_for_quoted_text ("-fpack-struct").get (),
-		"gcc/Code-Gen-Options.html#index-fpack-struct");
+  const char *s1 = u.get_url_suffix_for_quoted_text ("-fpack-struct").get ();
+  ASSERT_TRUE (!strcmp (s1,
+			"gcc/Code-Gen-Options.html#index-fno-pack-struct")
+	       || !strcmp (s1,
+			   "gcc/Code-Gen-Options.html#index-fpack-struct"));
 
   /* Check a "-fno-" variant of an option.  */
-  ASSERT_STREQ (u.get_url_suffix_for_quoted_text ("-fno-inline").get (),
-		"gcc/Optimize-Options.html#index-finline");
+  const char *s2 = u.get_url_suffix_for_quoted_text ("-fno-inline").get ();
+  ASSERT_TRUE (!strcmp (s2,
+			"gcc/Optimize-Options.html#index-fno-inline")
+	       || !strcmp (s2,
+			   "gcc/Optimize-Options.html#index-finline"));
 }
 
 /* Run all of the selftests within this file.  */
-- 
2.52.0

